pkg/interface/chat/src/js/components/settings.js

```diff
@@ -234,6 +234,13 @@ export class SettingsScreen extends Component {
         text = "Leaving...";
       }
 
+      let title = props.station.substr(1);
+
+      if ((props.association) && (props.association.metadata)) {
```

Comment:
“metadata“ in props.associations for safety

Reply:
Wait, I don't follow why this one isn't safe. The && won't hit the second check unless the first one is initialised at all, and this is just a property check, so it won't hit `can't find metadata of undefined`

Reply:
Basically the pattern I use does assume it's being passed an object — and if it isn't, it'll probably crash. And I can see why the `in` operator catches that. But I can't think of a situation where it wouldn't be either uninitialised or an object...hm. I will do it anyway but want to figure out the rationale for safety here

Reply:
At this point it’s just an pre-rational embedded heuristic I use as a means of overly defensive programming within a hostile JS environment, I’ll try to find resources relating to it.

Reply:
Alright, I’ve come to the conclusion that I just have an irrational fear of object accessors, I can’t find anything.

Reply:
I will do it anyway

Reply:
Potentially of interest: [`?.`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

(not suggesting using this right now, of course. just sharing)

Reply:
I’m aware of option chaining but I’m unsure if we use Babel and iirc native support is patchy.

Reply:
?. is a lovely convention in swift, hopefully it'll become widely used in JS.